shell/browser/ui/win/notify_icon_host.cc

```diff
@@ -162,15 +163,30 @@ LRESULT CALLBACK NotifyIconHost::WndProc(HWND hwnd,
 
     switch (lparam) {
       case NIN_BALLOONSHOW:
-        win_icon->NotifyBalloonShow();
+        content::GetUIThreadTaskRunner({})->PostTask(
+            FROM_HERE, base::BindOnce(
+                           [](base::WeakPtr<NotifyIcon> wicon) {
+                             wicon->NotifyBalloonShow();
+                           },
+                           win_icon));
```

Comment:
The code as is will fail because it doesn't check `if (wicon)`. I recommend using the member-function variant of `base::Bind` instead, which handles this automatically:

```suggestion
        content::GetUIThreadTaskRunner({})->PostTask(
            FROM_HERE, base::BindOnce(
                           &NotifyIcon::NotifyBalloonShow,
                           win_icon));
```

(see [base/callback.h:46-49](https://source.chromium.org/chromium/chromium/src/+/master:base/callback.h;l=46-49;drc=788a3897a44bde5263e83172c84ebb85d99b5080))

Same for the cases below.